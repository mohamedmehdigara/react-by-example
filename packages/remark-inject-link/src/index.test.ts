import { Emphasis, Text } from 'mdast'
import remark from 'remark'
import { Processor } from 'unified'
import remarkInjectLink, { Resolved } from '.'

const fixtures = {
  simple: `\
# Hello world!

## Heading _level_ 2

hello link _injector_
`,
}

describe('remarkInjectLink', () => {
  let processor: Processor

  beforeEach(() => {
    processor = remark().use(remarkInjectLink, {
      resolve({ type, children }: Emphasis): Resolved {
        if (type !== 'emphasis') return undefined

        const [{ value }] = children as Text[]
        return { text: value, url: `https://example.com/${value}` }
      },
    })
  })

  it.each(Object.entries(fixtures))(
    'should inject links "%s"',
    async (_name, markdown) => {
      const result = await processor.process(markdown)
      expect(result.toString()).toMatchSnapshot()
    },
  )
})
