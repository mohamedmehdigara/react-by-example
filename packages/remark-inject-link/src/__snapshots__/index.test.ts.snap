// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`remarkInjectLink should inject links "simple" 1`] = `
"# Hello world!

## Heading [level](https://example.com/level)_level_ 2

hello link [injector](https://example.com/injector)_injector_
"
`;
